### PR TITLE
docs: touchups for a better Lighthouse score

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   ~[ ] Addresses an existing issue: fixes #000~ 🤘
-   ~[ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)~
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md) were taken

## Overview

I ran a Lighthouse audit in Chrome on the production site. This fixes the reported complaints. Reasoning commented inline in this PR.

### CODE REVIEW

File: packages/website/docusaurus.config.js [hunk 1]
```diff
@@ -67,8 +67,9 @@ const config = {
         // hideOnScroll: true,
         logo: {
           alt: 'TypeScript ESLint',
+          height: '2rem',
           src: 'img/logo.svg',
-          // srcDark: 'img/logo-dark.svg',
+          width: '2rem',
```
Comment (JoshuaKGoldberg) [inline packages/website/docusaurus.config.js:72]: Images should have explicit height and width before they load. That way the browser knows how to resize them immediately.

Doesn't seem to be a real issue for us as the header is a fixed size but might as well improve for SEO 🤷

File: packages/website/src/css/custom.css [hunk 2]
```diff
@@ -31,6 +33,9 @@ html[data-theme='dark'] {
 
   --ifm-code-background: rgb(40, 42, 54);
   --ifm-code-color: rgb(248, 248, 242);
+  --ifm-color-info: var(--ifm-color-primary-light);
+  --ifm-menu-color-active: var(--ifm-color-primary-light);
+  --ifm-navbar-link-hover-color: var(--ifm-color-primary-light);
```
Comment (JoshuaKGoldberg) [inline packages/website/src/css/custom.css:38]: These were all accessibility color contrast issues.

File: packages/website/src/pages/index.tsx [hunk 3]
```diff
@@ -26,9 +26,9 @@ const features: FeatureItem[] = [
             />
           </div>
           <div className="col col--8">
-            <h4 className="text--justify">
+            <h3 className="text--justify">
               <b>ESLint</b> is an awesome linter for JavaScript code.
-            </h4>
+            </h3>
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/index.tsx:31]: Before, the page went `h1 > h2 > h4`. You can't jump up more than 1 heading level at a time.

https://dequeuniversity.com/rules/axe/4.0/heading-order

In theory I could have applied a CSS style to keep these at the same size as before, but I kinda like them slightly bigger now...

File: packages/website/src/pages/styles.module.css [hunk 4]
```diff
@@ -73,8 +73,10 @@
   border: none;
 }
 .tier-sponsor img {
+  max-height: 90px;
   width: 90px;
 }
 .tier-supporter img {
+  max-height: 60px;
```
Comment (JoshuaKGoldberg) [inline packages/website/src/pages/styles.module.css:80]: Same image sizing request as before.

**Comment by Josh-Cena** [comment]
image

🧐I suggest always use `px` for HTML markup

**Comment by JoshuaKGoldberg** [comment]
Haha, I swear that was regularly sized on local... switching now to `32px`.

> `px`

Funny, I've been trained to always use `rem` specifically.

**Comment by Josh-Cena** [comment]
I'm not savvy with web standards, but I have a feeling that we may have given too much freedom in the Docusaurus API... https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img#attr-width If we ever go back (which was just a few weeks ago) we should just allow numbers.

`rem` is better in CSS but not in HTML markup

**Review by bradzacher — APPROVED** [review]

**Pull request merged**